src/current_user.ts

```diff
@@ -394,36 +429,51 @@ export default class CurrentUser {
     this.typingStateChange(eventPayload, roomId, onSuccess, onError);
   }
 
-  addMessage(
-    text: string,
-    room: Room,
+  sendMessage(
+    options: SendMessageOptions,
     onSuccess: (messageId: number) => void,
     onError: (error: any) => void,
   ) {
-    const messageObject = {
-      text,
-      user_id: this.id,
-    };
+    const { linkAttachment, dataAttachment, ...rest } = options;
 
-    this.instance
-      .request({
-        json: messageObject,
-        method: 'POST',
-        path: `/rooms/${room.id}/messages`,
-      })
-      .then((res: any) => {
-        const messageIdPayload = JSON.parse(res);
-        const messageId = messageIdPayload.message_id;
-        // TODO: Error handling
-        onSuccess(messageId);
-      })
-      .catch((error: any) => {
-        this.instance.logger.verbose(
-          `Error adding message to room ${room.name}:`,
-          error,
+    if (linkAttachment !== undefined && dataAttachment !== undefined) {
+      onError(
+        new Error('You can only provide at most one attachment per message'),
+      );
+      return;
+    }
+
+    if (dataAttachment !== undefined) {
+      const { file, name } = dataAttachment;
+      this.uploadFile(file, name, options.roomId).then((fileRes: any) => {
+        this.sendMessageWithCompleteOptions(
+          {
+            attachment: fileRes,
+            user_id: this.id,
+            ...rest,
+          },
+          onSuccess,
+          onError,
         );
-        onError(error);
       });
+    } else {
+      // TODO: Shouldn't be an any
```

Comment:
```
      const completeOptions: CompleteMessageOptions = {
        user_id: this.id,
        ...rest,
      };
```
?

Reply:
I think I had that a while back and then undid too much and lost it. Thanks!